### DISCUSSION TITLE
Suggestion: Disable-HTTPS environment variable

### DISCUSSION THREAD

**Discussion by UltraBlackLinux** [dani-garcia/vaultwarden#4626] [Ideas]
I wanted to get this set up, but I'm really not comfortable with hiding all my passwords behind two lousy passwords, like the duckdns thing suggests. I also don't want vaultwarden to be visible on my dns record or anything like that.
Instead I'd like to run this purely locally, without HTTPS, behind a wireguard VPN. I don't think a lot can really go wrong here apart from my router getting hacked.
I cannot use vaultwarden like this because it wants HTTPS and I cannot get a certificate working here without substantial hassle. 
As such, I'd like to request a "disable HTTPS" environment variable.
Is that possible?
Thanks!

**Comment by BlackDex** [comment]
No, that is not possible if you want to connect to it from anything else but `localhost`/`127.0.0.1`. The browsers do not support the encryption library in those cases. So either create a port-forward/proxy from your local device to your Vaultwarden instance so it can be accessible via localhost, or create a self-signed certificate with all the hassle that comes with it.

You can also for example create a wildcard certificate for a (sub)domain with LetsEncrypt. That way nobody needs to know any sub domain name, but that does require an owned domain.

**Reply by UltraBlackLinux** [reply]
hmm, Unfortunately I'm not terribly happy with either of those solutions.

I guess the wildcard certificate is my best guess but I don't know if caddy can do that. I'll have to check.

Thanks for your help!

**Reply by kpfleming** [reply]
Keep in mind that these restrictions are the result of the behavior of browsers and the Bitwarden clients; Vaultwarden cannot change those restrictions. The same would apply if you wanted to self-host Bitwarden without HTTPS.
